src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,98 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] storage;
+    private int size;
+
+    public ArrayList() {
+        this.storage = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkArraySizeAndResize();
+        size++;
+        checkIndex(index);
+        System.arraycopy(storage, index, storage, index + 1, size - 1 - index);
+        storage[index] = value;
```

Comment:
```suggestion
        checkArraySizeAndResize();
        checkIndex(index);
        System.arraycopy(storage, index, storage, index + 1, size - index);
        storage[index] = value;
        size++;
```
Shortly in this case.

Reply:
варто залишити мій варіант
інакше прийдеться переробити метод checkIndex